sale_import_base/tests/test_sale_order_import.py

```diff
@@ -0,0 +1,191 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from copy import deepcopy
+
+from odoo.exceptions import ValidationError
+
+from .common_sale_order_import import SaleImportCase
+
+
+class TestSaleOrderImport(SaleImportCase):
+    def setUp(self):
+        super().setUp()
+
+    def test_invalid_json(self):
+        """ An invalid input will stop the job """
+        json_import = deepcopy(self.sale_order_example_vals)
+        del json_import["address_customer"]["street"]
+        del json_import["address_invoicing"]["zip"]
+        del json_import["address_shipping"]["country_code"]
+        json_import["lines"][0] = self.line_invalid
+        with self.assertRaises(ValidationError):
+            self.env["sale.order"].process_json_import(json_import)
+
+    def test_create_partner(self):
+        """ If we can't match a partner on 1. external id
+        or 2. email (optional), create a new one. """
+        json_import = deepcopy(self.sale_order_example_vals)
+        json_import["address_customer"] = {
+            "name": "Nicolas Cage",
+            "street": "1",
+            "street2": "bis",
+            "zip": "22121",
+            "city": "New Orleans",
+            "email": "nicolas.cage@aol.com",
+            "country_code": "US",
+            "external_id": "AMZN_NICCAGE",
+        }
+        json_import["address_invoicing"] = {
+            "name": "Nicolas Cage",
+            "street": "2",
+            "street2": "bis",
+            "zip": "22121",
+            "city": "Somerset",
+            "country_code": "GB",
+        }
+        json_import["address_shipping"] = {
+            "name": "Nicolas Cage",
+            "street": "3",
+            "street2": "bis",
+            "zip": "22121",
+            "city": "Oberpfaltz",
+            "country_code": "DE",
+        }
+        self.env["sale.order"].process_json_import(json_import)
+        customer = self.env["res.partner"].search([], order="id desc")[2]
```

Comment:
this will work randomly depending on the existing partner of the database

Reply:
Also instead of using a full new json, use the original

Reply:
sale = self.env["sale.order"].process_json_import(json_import)

customer = sale.partner_id

Reply:
count_initial_partner = self.env["res.partner"].search([("parent_id", "=", False)])


count_partner = self.env["res.partner"].search([("parent_id", "=", False)])
self.assertEqual(count_partner, count_initial_partner +1)